core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4705,6 +4705,29 @@ public DependencyGraph getDependencyGraph() {
     public List<ManagementLink> getManagementLinks() {
         return ManagementLink.all();
     }
+
+    // for Jelly
+    @Restricted(NoExternalUse.class)
+    public Map<ManagementLink.Category, List<ManagementLink>> getCategorizedManagementLinks() {
+        Map<ManagementLink.Category, List<ManagementLink>> byCategory = new TreeMap<>();
+        for (ManagementLink link : ManagementLink.all()) {
+            if (link.getIconFileName() == null) {
+                continue;
+            }
+            if (!Jenkins.get().hasPermission(link.getRequiredPermission())) {
+                continue;
+            }
+            String categoryName = link.getCategory();
+            ManagementLink.Category category;
+            try {
+                category = ManagementLink.Category.valueOf(categoryName);
+            } catch (IllegalArgumentException ex) {
```

Comment:
need to catch nullPointerException unless you switch to my proposed way (as there are no annotations for plugins not upgrading core and they could heaven forbid return null without any compiler warnng)